include/cetl/pmr/interface_ptr.hpp

```diff
@@ -0,0 +1,92 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef CETL_PMR_INTERFACE_PTR_H_INCLUDED
+#define CETL_PMR_INTERFACE_PTR_H_INCLUDED
+
+#ifndef CETL_H_ERASE
+#    include "cetl/cetl.hpp"
+#endif
+
+#include "cetl/unbounded_variant.hpp"
+
+#include <memory>
+#include <functional>
```

Comment:
This is temporary (see below `std::function<void(Interface*)> deleter_;`) - will be replaced with PMR `cetl::pmr::function` when it will be available.

Reply:
Note for Scott & Lydia: `cetl::function` is a near-term goal that we are now working on.

Reply:
yes, I already have my private `sshirokov/84_pmr_function` branch with very first draft of implementation.